src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminAuditController.java

```diff
@@ -0,0 +1,67 @@
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.dts.fact.config.security.Role;
+import uk.gov.hmcts.dts.fact.model.admin.Audit;
+import uk.gov.hmcts.dts.fact.services.admin.AdminAuditService;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.http.ResponseEntity.ok;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_ADMIN;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@RestController
+@RequestMapping(
+    path = "/admin/audit",
+    produces = {MediaType.APPLICATION_JSON_VALUE}
+)
+@Slf4j
+public class AdminAuditController {
+    private final AdminAuditService adminAuditService;
+
+    @Autowired
+    public AdminAuditController(AdminAuditService adminAuditService) {
+        this.adminAuditService = adminAuditService;
+    }
+
+    @GetMapping(params = {"page", "size"})
+    @ApiOperation("Find all audits based on the provided parameters.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successful", response = Audit.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message = "Forbidden")
+    })
+    @Role({FACT_SUPER_ADMIN})
+    public ResponseEntity<List<Audit>> getAudits(@RequestParam("page") int page,
+                                                 @RequestParam("size") int size,
+                                                 @RequestParam(value = "location", required = false) Optional<String> location,
+                                                 @RequestParam(value = "email", required = false) Optional<String> email,
+                                                 @RequestParam(value = "date-from", required = false)
+                                                     @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) Optional<LocalDateTime> dateFrom,
+                                                 @RequestParam(value = "date-to", required = false)
+                                                     @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) Optional<LocalDateTime> dateTo) {
+
+        // Return with an error if one or the other is set, but not both
+        if (dateFrom.isPresent() && dateTo.isEmpty() || dateFrom.isEmpty() && dateTo.isPresent()) {
```

Comment:
Is this an error? Could a missing 'to' or 'from' mean 'give me all from this date forward' or 'give me all up to this date'?

Reply:
Its "select all the audits from this date, to this other date". So if we have a "from this date" but not a "to this date", we should return with a bad request

For example 

localhost:8080/admin/audit?page=0&size=50&location=birmingham-civil-and-family-justice-centre&email=hmcts.super.fact@gmail.com&date-from=2021-09-06T12:09:42.411&date-to=

Rather than:

localhost:8080/admin/audit?page=0&size=50&location=birmingham-civil-and-family-justice-centre&email=hmcts.super.fact@gmail.com&date-from=2021-09-06T12:09:42.411&date-to=2022-01-13T17:09:42.411

Reply:
Marking as resolved - this was just some food for thought. We could allow a from or to date to be missing and just treat that part as open-ended.